### PR TITLE
feat: should allow `options` http method

### PR DESCRIPTION
**Pull request by ghost**
Reason：According to the bug report from https://github.com/eggjs/egg/issues/2866, it seems we should allow `OPTIONS` as a proper way.

Change List:

1) `methodnoallow.js`：Remove `OPTIONS`.
2) `method_not_allow.test.js`：Remove `OPTIONS`.
3) Change `README.md` in both English and Chinese versions.

### CODE REVIEW

**Comment by Maledong** [comment]
@fengmk2：It passed v8.x on Nodejs but failed in v10.x？And I've ran successfully on my machine.

**Comment by Maledong** [comment]
@fengmk2 ：I tested in both v8.x and v10.x (all of the are from https://nodejs.org/zh-cn/) and everything goes well with me, but I don't know why it failed on the CI test?

**Comment by Maledong** [comment]
PS：The reason why I changed from `safeHttpMethodsMap` (a map-based JSON obj) to `Set` is based on my benchmark test (Mocking):

```
const a = {};
const set = new Set(['trace','track']);

a['trace'] = true;
a['track'] = true;

const BenchTest = require('benchmark');
const suite = new BenchTest.Suite();

suite
    .add("indexOf", () => {
        a["F".toLowerCase()];
    })
    .add("Map", () => {
        set.has("F".toLowerCase());
    })
    // add listeners
    .on('cycle', function (event) {
        console.log(String(event.target));
    })
    .on('complete', function () {
        console.log('Fastest is ' + this.filter('fastest').map('name'));
    })
    // run async
    .run({ 'async': true });
```
The result is:
mapfast

So for looking for a certain value in a collection. Map is much more faster.
And according to https://jsperf.com/find-by-map-with-find-by-array, it seems JSON's object mapping is slower than indexOf.

catch

**Comment by ghost** [comment]
@dead-horse：`OPTIONS` is a normal submitting way, so this shouldn't be removed :)

File: lib/middlewares/methodnoallow.js [hunk 1]
```diff
@@ -1,10 +1,11 @@
 'use strict';
 
 const methods = require('methods');
-
+const METHODS_NOT_ALLOWED = [ 'trace', 'track' ];
 const safeHttpMethodsMap = {};
+
 for (const method of methods) {
-  if ([ 'trace', 'track', 'options' ].indexOf(method) === -1) {
+  if (METHODS_NOT_ALLOWED.indexOf(method) === -1) {
```
Comment (fengmk2) [inline lib/middlewares/methodnoallow.js]: use `includes` because we only support node >= 8

File: test/method_not_allow.test.js [hunk 2]
```diff
@@ -59,11 +59,4 @@ describe('test/method_not_allow.test.js', () => {
       .set('accept', 'text/html')
       .expect(405);
   });
-
-  it('should not allow option method', () => {
```
Comment (fengmk2) [inline test/method_not_allow.test.js]: keep this test and make sure we support option method.

**Review by fengmk2 — APPROVED** [review]

File: test/method_not_allow.test.js (hunk 2 above)
Comment (ghost) [reply test/method_not_allow.test.js]: Fixed but change to `should allow`.

File: lib/middlewares/methodnoallow.js (hunk 1 above)
Comment (ghost) [reply lib/middlewares/methodnoallow.js]: OK

**Comment by ghost** [comment]
Fixed, @fengmk2. Thanks!

**Review by atian25 — APPROVED** [review]

**Comment by atian25** [comment]
need rebase

**Comment by ghost** [comment]
@atian25：OK

**Review by dead-horse — APPROVED** [review]

**Comment by ghost** [comment]
Thanks all!!!

**Pull request merged**